File: internal/api/dto/Task.go
package dto

import "time"

type TaskCreate struct {
	Tag         string `json:"tag"`
	Sprint      int    `json:"sprint"`
	Description string `json:"description"`
}

type TaskUpdate struct {
	Tag         string    `json:"tag"`
	Sprint      int       `json:"sprint"`
	Description string    `json:"description"`
	Completed   bool      `json:"completed"`
	StartAt     time.Time `json:"start_at"`
	EndAt       time.Time `json:"end_at"`
}

File: internal/presentation/react/src/domains/Task.ts
import { User } from "./User";

export enum Completed {
    INCOMPLETED,
    COMPLETED,
    IGNORED,
}

export type Task = {
    id: Number;
    tag: String;
    sprint: String;
    description: String;
    completed: Boolean;
    start_at: Date;
    end_at: Date;
    user: User;
};

export type TaskCreate = {
    tag: String;
    sprint: Number;
    description: String;
    completed: Boolean;
};

export type TaskFilter = {
    tag?: String;
    completed?: Completed;
    sprints?: Number[];
    start_at?: Date;
    end_at?: Date;
};
